vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1478,15 +1507,8 @@ func resourceVSphereComputeClusterApplyVsanConfig(d *schema.ResourceData, meta i
 		},
 	}
 
-	dedupEnabled := d.Get("vsan_dedup_enabled").(bool)
-	compressionEnabled := d.Get("vsan_compression_enabled").(bool)
-	if dedupEnabled && !compressionEnabled {
-		return fmt.Errorf("vsan compression must be enabled if vsan dedup is enabled")
-	}
-
-	conf.DataEfficiencyConfig = &vsantypes.VsanDataEfficiencyConfig{
-		DedupEnabled:       dedupEnabled,
-		CompressionEnabled: &compressionEnabled,
+	if version.AtLeast(viapi.VSphereVersion{Product: version.Product, Major: 8, Minor: 0}) {
```

Comment:
I won't hold the PR up on this though if things are working since this has been a prioritized issue, let me have a short window to see if there's SOMETHING else that could be done to version gate this.

Reply:
I will allow the old school version gating for vSphere 8 for now to not be a blocker.